### ISSUE TITLE
ppc64le: Infer --without-sse on configure

### ISSUE THREAD

**Issue by jlost** [gmarcais/Jellyfish#64]
Jellyfish should build cleanly on PPC64le by programmatically determining the need for --without-sse.

See #63 for details.

**Comment by rafaeldelucena** [comment]
Hi,

I had some research about macros on autotools and i found this:

https://www.gnu.org/software/autoconf-archive/ax_ext.html

**Comment by edelsohn** [comment]
What result with the proposed increased bounty amount accomplish?

**Comment by rafaeldelucena** [comment]
#68

**Comment by rafaeldelucena** [comment]
I'm some problems running a test infrastructure for PPC64, using it [PPC64 Debian](https://wiki.debian.org/ppc64el/Installation) as a base,  lack of packages, problems in network configuration .. if you have any vm available, the work would be finalized cheaper and faster.

**Comment by edelsohn** [comment]
http://osuosl.org/services/powerdev

**Comment by edelsohn** [comment]
We have increased the bounty.

**Comment by rafaeldelucena** [comment]
Hello, I had some news!

Based on my tests, the addition of these macros and the change in configure.ac should be enough to enable SSE only when detected on the host machine.

Take a look at https://github.com/gmarcais/Jellyfish/pull/68

**Comment by rafaeldelucena** [comment]
The [build result for PPC64LE](https://gist.github.com/rafaeldelucena/81f9b0befd2c0b24eb2c8c7fcef43f8b)

**Comment by dmiller423** [comment]
It appears there were multiple related issues raised for this :|  see #63 #64 
I didn't know you were working on this,  and probably autoconfig tests

**Comment by rafaeldelucena** [comment]
Hello @dmiller423,

I think #63 it's a major issue with #64 and #65 as child tasks, I'm still waiting for the review of the developers, also from @edelsohn, this project seems to be idle at this moment.

**Comment by edelsohn** [comment]
I'm waiting for the maintainers to pull the patches and declare the issue closed.

**Comment by dmiller423** [comment]
Yea I didn't want step on anyones toes,  i didn't see there were 3 different bounties involved for the same thing :|

**Comment by jlost** [comment]
The two for this project are distinct -- one for fixing the tests that fail erroneously, and one for improving the build system (this one). What's the third?

**Comment by dmiller423** [comment]
https://www.bountysource.com/issues/36025820-ppc64le-make-fails-on-jellyfish-need-without-sse
https://www.bountysource.com/issues/36403911-ppc64le-infer-without-sse-on-configure
https://www.bountysource.com/issues/36404265-ppc64le-unit-test-fails-erroneously

**Comment by dmiller423** [comment]
They are all the same problem,  except the bit about __m128 test raphael commented on.

**Comment by jlost** [comment]
The Jellyfish issue #64 would also fix the Sailfish issue, but since there was no guarantee the Jellyfish fix would be worked on or accepted, the Sailfish issue was intended to at least provide a workaround to build Sailfish on ppc64le.

**Comment by gmarcais** [comment]
Hi, I am a little behind on that thread. Can someone fill me in on the status of those patches? I am definitely interested in having the detection of SSE being automated and ready to pull in those fixes, if they work.

**Comment by rafaeldelucena** [comment]
Hello,

My solution add some open source m4 macros from GNU repository, detecting and fill the defines for each detected Vector instructions from various architectures, so will be useful if you plan to add optimizations to other platforms or to newer versions of SSE.

The David solution is smaller and simpler, it's more focused for this issue I think, but if you plan to add new kind of optimizations, write more tests on configure.ac can start to become a mess.

**Comment by dmiller423** [comment]
@gmarcais  the pull request I issued see: https://github.com/gmarcais/Jellyfish/pull/80
Is very simple,  it does a quick compile check for SSE and sets a default for your --with-sse argument.

@rafaeldelucena I also initially looked at AX_EXT but I found it extremely lacking,  it had one check for altivec that didn't even work on either of my vm's and required an additional m4 file that autoreconf won't add by itself.  It would be the right solution, but someone would need to fix it up and push it upstream to the m4 ext repo.

**Comment by gmarcais** [comment]
Alright, I'll try to look at those pull request by the end of the week. No sooner though...

**Comment by rafaeldelucena** [comment]
@dmiller423 

> it had one check for altivec that didn't even work on either of my vm's

I've already update the m4 macro to detect Altivec and VSX on newer machines :o)

> it failed to build because of x86 defines and required an additional m4 file that autoreconf won't add by itself. 

The required macros already have been included on the PR not only AX_EXT, I don't understand this statement.

> It would be the right solution, but someone would need to fix it up and push it upstream to the m4 ext repo.

Waiting for https://savannah.gnu.org/patch/index.php?9237

@gmarcais
Ok, I'm going to cheer to get feedback from the GNU community until there, to at least leave you with doubts in which version to choose. :P

**Comment by edelsohn** [comment]
The Altivec and VSX bits really are about hardware resources -- the Altivec registers and the VSX registers.  POWER7, POWER8 and POWER9 add more VSX instructions to the ISA, but that probably depends on the architecture level targeted by the toolchain and its options.

**Comment by rafaeldelucena** [comment]
The m4 update patch was accepted :)

**Comment by barkovv** [comment]
@rafaeldelucena What's the status of this issue?

**Comment by rafaeldelucena** [comment]
@barkovv the job is done, waiting for @gmarcais decision for what PR will be integrated on upstream.

**Comment by rafaeldelucena** [comment]
Hello @gmarcais any thoughts about the solution for this issue?

[]s

**Comment by rafaeldelucena** [comment]
Still waiting for review :'(

**Comment by gmarcais** [comment]
Actually, the current revision Has this patch applied. See commits 69729191f85aaba488cb6e903f743f41c4a0449c and 890ed7abd9b8660246d40c139b8eae54d979f8a6. It is in version 2.2.7.

**Comment by rafaeldelucena** [comment]
The patch accepted is using an older version of the ax_ext.m4 which doesn't work for newer platforms, I've updated and have been accepted at GNU archive:

https://www.gnu.org/software/autoconf-archive/ax_ext.html
https://github.com/peti/autoconf-archive/blob/master/m4/ax_ext.m4

and rebased the PR #68 

:)

**Comment by rafaeldelucena** [comment]
PR #68 closed because of merge in develop 958f049

**Comment by rafaeldelucena** [comment]
I tested the develop branch on a ppc64le platform, with `./configure --without-int128`, and the tests passed.

```
Linux jellyfish-1 3.16.0-4-powerpc64le #1 SMP Debian 3.16.7-ckt25-2+deb8u3 (2016-07-02) ppc64le GNU/Linux
```

```
debian@jellyfish-1:~/JellyFish$ make check
make  check-am
make[1]: Entering directory '/home/debian/JellyFish'
make  libgtest.la libgtest_main.la bin/generate_sequence bin/fastq2sam bin/test_all
make[2]: Entering directory '/home/debian/JellyFish'
make[2]: 'libgtest.la' is up to date.
make[2]: 'libgtest_main.la' is up to date.
make[2]: 'bin/fastq2sam' is up to date.
make[2]: 'bin/test_all' is up to date.
make[2]: Leaving directory '/home/debian/JellyFish'
make  check-TESTS
make[2]: Entering directory '/home/debian/JellyFish'
make[3]: Entering directory '/home/debian/JellyFish'
PASS: tests/generate_sequence.sh
PASS: tests/parallel_hashing.sh
PASS: tests/merge.sh
PASS: tests/bloom_filter.sh
SKIP: tests/big.sh
PASS: tests/subset_hashing.sh
PASS: tests/multi_file.sh
PASS: tests/bloom_counter.sh
PASS: tests/large_key.sh
SKIP: tests/sam.sh
PASS: tests/small_mers.sh
SKIP: tests/swig_python.sh
SKIP: tests/swig_ruby.sh
SKIP: tests/swig_perl.sh
PASS: unit_tests/unit_tests.sh
make[4]: Entering directory '/home/debian/JellyFish'
make  all-am
make[5]: Entering directory '/home/debian/JellyFish'
make[5]: Leaving directory '/home/debian/JellyFish'
make[4]: Leaving directory '/home/debian/JellyFish'
============================================================================
Testsuite summary for jellyfish 2.2.8
============================================================================
# TOTAL: 15
# PASS:  10
# SKIP:  5
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
make[3]: Leaving directory '/home/debian/JellyFish'
make[2]: Leaving directory '/home/debian/JellyFish'
make[1]: Leaving directory '/home/debian/JellyFish'
```

For me this issue is resolved, could you close it?
